subscription_manager.go

```diff
@@ -553,6 +562,19 @@ func SubscriptionWithMessageTimeToLive(window *time.Duration) SubscriptionManage
 	}
 }
 
+// SubscriptionWithDefaultRuleDescription configures the subscription to set a
+// default rule
+func SubscriptionWithDefaultRuleDescription(filter FilterDescriber) SubscriptionManagementOption {
+	return func(s *SubscriptionDescription) error {
+		rule := &DefaultRuleDescription{
+			Filter: filter.ToFilterDescription(),
+			Name:   ptrString("rule0"),
```

Comment:
Is this name correct?  I did a little searching and all I can find is `$Default`.  Should that be used instead?

Reply:
It was changed from `$Default` to `rule0` in [d452ef9](https://github.com/Azure/azure-service-bus-go/pull/172/commits/d452ef94455eae1e3822ef47413cfcdbfa512e15). For some reason when using the name `$Default` the filter is not set, so an empty filter is created. (`$` is actually an invalid character in an entity name, so it shouldn't work at all)

I can change the name to `_Default` or add a name field to `SubscriptionWithDefaultRuleDescription` so the caller can set an arbitrary name. Which option do you think would be most consistent/preferred?

Reply:
I see, thanks for clarifying.  Mostly I wasn't sure if the name can be arbitrary.  If it can be user-defined I like your second option of providing a way to configure it but falling back to some default (I do like `_Default` too).

Reply:
Cool, I've added user-defined rule name to now. I think a fall back to `_Default` would just be misleading and prefer the user to get an error from service bus explaining what went wrong (incorrect character in name or whatever).